Source/items.cpp

```diff
@@ -1373,40 +1374,41 @@ int AllocateItem()
 	return inum;
 }
 
-static void GetSuperItemSpace(int x, int y, int8_t inum)
+static void GetSuperItemSpace(Point pos, int8_t inum)
 {
-	if (!GetItemSpace(x, y, inum)) {
-		for (int k = 2; k < 50; k++) {
-			for (int j = -k; j <= k; j++) {
-				int yy = y + j;
-				for (int i = -k; i <= k; i++) {
-					int xx = i + x;
-					if (ItemSpaceOk(xx, yy)) {
-						items[inum].position = { xx, yy };
-						dItem[xx][yy] = inum + 1;
-						return;
-					}
-				}
+	if (GetItemSpace(pos, inum))
+		return;
+	for (int k = 2; k < 50; k++) {
+		for (int j = -k; j <= k; j++) {
+			int yy = pos.y + j;
+			for (int i = -k; i <= k; i++) {
+				int xx = i + pos.x;
+				if (!ItemSpaceOk(xx, yy))
+					continue;
+				items[inum].position = { xx, yy };
+				dItem[xx][yy] = inum + 1;
+				return;
 			}
 		}
 	}
 }
 
-void GetSuperItemLoc(int x, int y, int *xx, int *yy)
+Point GetSuperItemLoc(Point pos)
 {
-	int i, j, k;
+	Point ret;
 
-	for (k = 1; k < 50; k++) {
-		for (j = -k; j <= k; j++) {
-			*yy = y + j;
-			for (i = -k; i <= k; i++) {
-				*xx = i + x;
-				if (ItemSpaceOk(*xx, *yy)) {
-					return;
+	for (int k = 1; k < 50; k++) {
+		for (int j = -k; j <= k; j++) {
+			ret.y = pos.y + j;
+			for (int i = -k; i <= k; i++) {
+				ret.x = i + pos.x;
+				if (ItemSpaceOk(ret.x, ret.y)) {
+					return ret;
 				}
 			}
 		}
 	}
+	return ret;

```

Comment:
This would probably be a fail condition. But not sure how we should express it, app_fatal seams a bit drastic.

Reply:
Maybe an assert?

Reply:
That might work, but it's a bit tricky since it's a state that is easily manipulated by the player.

Reply:
Agreed. Maybe the end-result is that some other item on the ground is overwritten.